### PR TITLE
Add Empty State, a Button and Conversion Tracking to the sample extension

### PR DESCRIPTION
**Pull request by IvanAgCa**
# Ticket

[CN-7417](https://unbounce.atlassian.net/browse/CN-7417)

# List of changes

- New configurable button that can be stylable:

image

- Empty state of the app:

image

- Convertion tracking (you can test this in a published page):

image

### CODE REVIEW

File: src/hello-world/components/hello-world.tsx [hunk 1]
```diff
@@ -1,32 +1,82 @@
...
+};
 
-const HelloWorld = ({ data, dispatch, className, mode }: ComponentProps<DataStructure, WithStylesProps>) => {
-  const { firstName, lastName } = data;
+const HelloWorld = ({ data, dispatch, className, mode, entityId }: ComponentProps<DataStructure, WithStylesProps>) => {
+  const { fullname, isButtonSet, tracking } = data;
   const [showModal, setShowModal] = useState(false);
+  const buttonId = `myAppButton-${entityId}`;
 
   const updateFirstName = (newFirstName: string) => {
```
Comment (pabloaredu) [inline src/hello-world/components/hello-world.tsx:24]: I would change the name of the function to `updateFullName` and also the parameter to `newFullName` for consistency.

File: src/hello-world/components/hello-world.tsx [hunk 2]
```diff
@@ -1,32 +1,82 @@
...
-    dispatch((api) => api.get('firstName').set(newFirstName));
+    dispatch((api) => api.get('fullname').set(newFirstName));
```
Comment (pabloaredu) [inline src/hello-world/components/hello-world.tsx:25]: same here, would be `newFullName`

File: src/hello-world/components/hello-world.tsx [hunk 3]
```diff
@@ -1,32 +1,82 @@
...
+                  button.onclick=function () { 
+                    ${getAfterFormSubmitScript(buttonId, true)}
+                  }
+                }())
+              `,
+              }}
+            />
+          ) : null}
+          {showModal && (
+            <ChangeFirstNameModal fullname={fullname} onUpdate={updateFirstName} onClose={() => setShowModal(false)} />
```
Comment (pabloaredu) [inline src/hello-world/components/hello-world.tsx:63]: Same comment as above. Use `FullNameModal` and `updateFullName` for consistency.

File: src/hello-world/components/hello-world.tsx (hunk 1 above)
Comment (IvanAgCa) [reply src/hello-world/components/hello-world.tsx:24]: Sure! I will do the change 👌🏻

**Review by pabloaredu — APPROVED** [review]
LGTM!

**Pull request merged**